plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -262,8 +262,18 @@ func (m *MQTTConsumer) onMessage(_ mqtt.Client, msg mqtt.Message) {
 
 	metrics, err := m.parser.Parse(msg.Payload())
 	if err != nil {
-		msg.Ack()
+		if m.PersistentSession {
+			msg.Ack()
+		}
 		m.acc.AddError(err)
+		<-m.sem
+		return
+	}
+	if len(metrics) == 0 {
+		if m.PersistentSession {
+			msg.Ack()
+		}
+		<-m.sem
 		return
 	}
```

Comment:
Can you maybe combine the two as `acc.AddError` will do nothing if `err` is `nil`!?

Reply:
ah thanks will do